drivers/mfd/mfd_tle9104.c

```diff
@@ -0,0 +1,682 @@
+	if (config->gpio_reset.port != NULL) {
+		if (!gpio_is_ready_dt(&config->gpio_reset)) {
+			LOG_ERR("%s: reset GPIO is not yet ready", dev->name);
+			return -ENODEV;
+		}
+
+		result = gpio_pin_configure_dt(&config->gpio_reset, GPIO_OUTPUT_ACTIVE);
+		if (result != 0) {
+			LOG_ERR("failed to initialize GPIO for reset");
+			return result;
+		}
+
+		k_busy_wait(TLE9104_RESET_DURATION_TIME_US);
+		gpio_pin_set_dt(&config->gpio_reset, 0);
+		k_busy_wait(TLE9104_RESET_DURATION_WAIT_TIME_US +
+			    TLE9104_RESET_DURATION_WAIT_TIME_SAFETY_MARGIN_US);
+	}
+
+	/*
+	 * The first read value should be the ICVID, this acts also as the setup of the
+	 * global status register address.
+	 */
+	result = tle9104_transceive_frame(dev, false, TLE9104REGISTER_GLOBALSTATUS, 0x00, &read_reg,
+					  &register_icvid);
+	if (result != 0) {
+		return result;
+	}
+
+	if (read_reg != TLE9104REGISTER_ICVID) {
+		LOG_ERR("expected to read register ICVID, got instead 0x%02X", read_reg);
+		return -EIO;
+	}
+
+	if (register_icvid != TLE9104_ICVERSIONID) {
+		LOG_ERR("got unexpected IC version id 0x%02X", register_icvid);
+		return -EIO;
+	}
+
+	result = tle9104_transceive_frame(dev, false, TLE9104REGISTER_GLOBALSTATUS, 0x00, &read_reg,
+					  &register_globalstatus);
+	if (result != 0) {
+		return result;
+	}
+
+	if (read_reg != TLE9104REGISTER_GLOBALSTATUS) {
+		LOG_ERR("expected to read register GLOBALSTATUS, got instead 0x%02X", read_reg);
+		return -EIO;
+	}
+
+	if ((register_globalstatus & TLE9104_GLOBALSTATUS_POR_LATCH_BIT) == 0) {
+		LOG_ERR("no power on reset detected");
+		return -EIO;
+	}
+
+	/* disable communication watchdog */
+	tle9104_set_register_bits(&register_cfg, TLE9104_CFG_CWDTIME_POS,
+				  TLE9104_CFG_CWDTIME_LENGTH, 0);
+
+	result = tle9104_write_register(dev, TLE9104REGISTER_CFG, register_cfg);
+	if (result != 0) {
+		LOG_ERR("unable to write configuration");
+		return result;
+	}
+
+	register_globalstatus = 0x00;
+	/* enable outputs */
+	register_globalstatus |= TLE9104_GLOBALSTATUS_OUTEN_BIT;
+
+	result = tle9104_write_register(dev, TLE9104REGISTER_GLOBALSTATUS, register_globalstatus);
+	if (result != 0) {
+		LOG_ERR("unable to write global status");
+		return result;
+	}
+
+	data->cwd_ignore = false;
+
+	register_cfg = 0x00;
+	switch (config->diagnostic_filter_time_us) {
+	case 2000:
+		tle9104_set_register_bits(&register_cfg, TLE9104_OFFDIAGCFG_DIAGFILTCFG_POS,
+					  TLE9104_OFFDIAGCFG_DIAGFILTCFG_LENGTH, 3);
+		break;
+	case 1000:
+		tle9104_set_register_bits(&register_cfg, TLE9104_OFFDIAGCFG_DIAGFILTCFG_POS,
+					  TLE9104_OFFDIAGCFG_DIAGFILTCFG_LENGTH, 2);
+		break;
+	case 500:
+		tle9104_set_register_bits(&register_cfg, TLE9104_OFFDIAGCFG_DIAGFILTCFG_POS,
+					  TLE9104_OFFDIAGCFG_DIAGFILTCFG_LENGTH, 1);
+		break;
+	case 200:
+		tle9104_set_register_bits(&register_cfg, TLE9104_OFFDIAGCFG_DIAGFILTCFG_POS,
+					  TLE9104_OFFDIAGCFG_DIAGFILTCFG_LENGTH, 0);
+		break;
+	default:
+		LOG_ERR("invalid value %i for diagnostic filter time",
```

Comment:
These are calculated at compile time, so this error message isn't needed.
You could calculate the value at compile time using DT_ENUM_IDX, then you wouldn't need the switch case.

Reply:
To be able to use DT_ENUM_IDX I would have to turn the enums into strings, if I understood it correctly. But these values actually represent numbers. Therefore it feels weird to turn them into strings.

I removed the default cases anyway, as they definitely won't be hit (unless there is a pointer going wild ...)

Reply:
DT_ENUM_IDX works with ints, e.g:
https://github.com/zephyrproject-rtos/zephyr/blob/f0b975032a68c2da38c34f01b836edb5e9e62b12/dts/bindings/pinctrl/ambiq%2Capollo4-pinctrl.yaml#L95-L106

Reply:
done